x-pack/plugins/alerts/server/task_runner/task_runner.ts

```diff
@@ -379,16 +375,10 @@ export class TaskRunner {
           return originalState;
         }
       ),
-      runAt: resolveErr<Date | undefined, Error>(runAt, (err) => {
-        return isAlertSavedObjectNotFoundError(err, alertId)
+      schedule: resolveErr<IntervalSchedule | undefined, Error>(schedule, (error) => {
+        return isAlertSavedObjectNotFoundError(error, alertId)
           ? undefined
```

Comment:
Ah, good catch... that's because we relied on an implicit behaviour there... because the task was not a scheduled task.
Considering we decided that scheduled tasks should keep running forever (from tTM's perspective) even if they fail and get deleted... so, arguably this is still correct.

What is missing, I guess, is a clean up process in the Alerting framework itself. I'll look into it.

Reply:
Addressed here: https://github.com/elastic/kibana/pull/80149/commits/16729cd4af9edb1df7aee595dce670a91cbc5b1a